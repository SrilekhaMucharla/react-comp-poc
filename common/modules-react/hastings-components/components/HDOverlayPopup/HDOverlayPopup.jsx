/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React, { useEffect, useState } from 'react';
// import './hd-overlay-popup.scss';
import { PropTypes } from 'prop-types';
import { createPortal } from 'react-dom';
import { Row, Col } from 'react-bootstrap';
import HDButtonRefactor from '../HDButtonRefactor/HDButtonRefactor';
import HDLabelRefactor from '../HDLabelRefactor/HDLabelRefactor';
import crossCircleBlack from '../../../../../applications/quote-and-buy/src/assets/images/icons/cross_circle_black.svg';

function HDOverlayPopup({
    children,
    id,
    hidden,
    confirmButton,
    cancelButton,
    onConfirm,
    onCancel,
    onBeforeOpen,
    onBeforeClose,
    showButtons,
    overlayButtonIcon,
    disabledConfirmButton,
    hideOnConfirm,
    labelText,
    customStyle,
    className,
    overlayButtonsClassName,
    closeOnSelect,
    headerBar
}) {
    const divToRender = document.getElementById('portal-root');
    const [modalShow, setModalShow] = useState(false);
    const [isStacked, setIsStacked] = useState(false);
    const [isVisibible, setIsVisible] = useState(false);
    const MODAL_CLOSE_DELAY = 500;

    useEffect(() => {
        if (closeOnSelect) {
            setModalShow(!closeOnSelect);
        }
    }, [closeOnSelect]);

    useEffect(() => {
        if (modalShow) {
            if (document.body.classList.contains('hd-modal-open')) {
                setIsStacked(true);
            } else {
                document.body.classList.add('hd-modal-open');
            }
            setTimeout(() => {
                setIsVisible(true);
            });
        } else if (!isStacked) {
            document.body.classList.remove('hd-modal-open');
        }
    }, [modalShow]);

    const handleCloseAnimation = (onAfterAnimation) => {
        setIsVisible(false);
        setTimeout(() => {
            if (typeof onAfterAnimation === 'function') {
                onAfterAnimation();
            }
            setModalShow(false);
        }, MODAL_CLOSE_DELAY);
    };

    const handleConfirm = (event) => {
        onConfirm(event);
        if (hideOnConfirm) {
            handleCloseAnimation();
        }
    };

    const handleCancel = (event) => {
        onCancel(event);
        handleCloseAnimation();
    };

    const handleClose = () => {
        handleCloseAnimation(onBeforeClose);
    };

    const handleClick = (event) => {
        if (typeof onBeforeOpen === 'function') {
            onBeforeOpen(event);
        }
        setModalShow(true);
    };

    const renderOverlay = () => (
        <div id={id} className={`hd-information-tooltip theme-white ${customStyle} ${className}`}>
            <div className="overlay">
                <div className="overlay-outside" onClick={handleClose} />
                <div className="overlay-container--outer">
                    <div className={`overlay-container${isVisibible ? ' is-visible' : ''}${headerBar ? ' overlay-container-multi' : ''}`}>
                        {headerBar}
                        <div className={`overlay-close${headerBar ? ' overlay-close-multi' : ''}`}>
                            <div onClick={handleClose}><img width={30} height={30} src={crossCircleBlack} alt="close" /></div>
                        </div>
                        {labelText && (
                            <div className="overlay-header">
                                <HDLabelRefactor Tag="h2" text={labelText} />
                            </div>
                        )}
                        <div className="overlay-body">{children}</div>
                        {showButtons && (
                            <div className="overlay-footer">
                                <Row>
                                    <Col xs={12} sm={6} className="pr-xs">
                                        <HDButtonRefactor
                                            variant="primary"
                                            className="theme-white"
                                            disabled={disabledConfirmButton}
                                            onClick={handleConfirm}
                                            label={confirmButton}
                                            fullWidth />
                                    </Col>
                                    <Col xs={12} sm={6} className="pl-xs overlay-footer__cancel-btn">
                                        <HDButtonRefactor
                                            variant="secondary"
                                            className="theme-white"
                                            onClick={handleCancel}
                                            label={cancelButton}
                                            fullWidth />
                                    </Col>
                                </Row>
                            </div>
                        )}
                    </div>
                </div>
            </div>
        </div>
    );

    return (
        <div className={`hd-information-tooltip ${className}`}>
            <div hidden={hidden} id={id} className={`hd-overlay-btn ${overlayButtonsClassName}`} onClick={handleClick}>
                {overlayButtonIcon}
            </div>
            {modalShow && createPortal(renderOverlay(), divToRender)}
        </div>
    );
}

HDOverlayPopup.propTypes = {
    children: PropTypes.node.isRequired,
    id: PropTypes.string.isRequired,
    hidden: PropTypes.bool,
    confirmButton: PropTypes.string,
    cancelButton: PropTypes.string,
    onConfirm: PropTypes.func,
    onCancel: PropTypes.func,
    onBeforeOpen: PropTypes.func,
    onBeforeClose: PropTypes.func,
    showButtons: PropTypes.bool,
    overlayButtonIcon: PropTypes.node,
    labelText: PropTypes.string,
    disabledConfirmButton: PropTypes.bool,
    hideOnConfirm: PropTypes.bool,
    customStyle: PropTypes.string,
    className: PropTypes.string,
    overlayButtonsClassName: PropTypes.string,
    closeOnSelect: PropTypes.bool,
    headerBar: PropTypes.shape({})
};

HDOverlayPopup.defaultProps = {
    hidden: false,
    confirmButton: 'Confirm',
    cancelButton: 'Cancel',
    onConfirm: () => { },
    onCancel: () => { },
    onBeforeOpen: () => { },
    onBeforeClose: () => { },
    showButtons: false,
    overlayButtonIcon: null,
    disabledConfirmButton: false,
    labelText: '',
    hideOnConfirm: true,
    customStyle: '',
    className: '',
    overlayButtonsClassName: '',
    closeOnSelect: false,
    headerBar: null
};

export default HDOverlayPopup;
