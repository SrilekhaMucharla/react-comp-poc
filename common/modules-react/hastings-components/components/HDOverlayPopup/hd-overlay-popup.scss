.hd-information-tooltip {
    .hd-overlay-btn {
        display: inline-flex;
        cursor: pointer;
    }

    h1, h2, h3, h4, h5 {
        color: $mono-midnight;
    }

    h5 {
        margin-top: $md-value;
        margin-bottom: 16px;
    }

    .overlay {
        position: fixed;
        left: 0;
        top: 0;
        width: 100%;
        height: 100%;
        background: $black-transparent-50;
        display: inline-flex;
        align-items: center;
        justify-content: flex-end;
        z-index: 1000000;

        .overlay-container {
            position: absolute;
            left: 100%;
            background-color: $white-color;
            padding: 60px 20px 40px 20px;
            width: 100%;
            min-height: 100%;
            border-radius: 0;

            @include media-breakpoint-up(md) {
                padding: 80px 65px 40px 65px;
            }

            /* Work around for no bottom padding on Firefox - Bugzilla issue id: 748518
            *  (applicable on older versions of Firefox, but including *newest* ESR ver.)
            *  Properties extracted from .overlay-container to prevent breaking existing page-level overrides.
            */
            &--outer {
                position: absolute;
                width: 100%;
                height: 100%;
                overflow-y: auto;
                overflow-x: hidden;

                @include media-breakpoint-up(sm) {
                    max-width: 375px;
                }
                @include media-breakpoint-up(md) {
                    width: 50%;
                    min-width: 559px;
                }
                @include media-breakpoint-up(xl) {
                    width: 30%;
                }
            }
        }

        .overlay-container-multi {
            padding: 100px 20px 40px 20px;
            @include media-breakpoint-up(md) {
                padding: 120px 65px 40px 65px;
            }
        }

        .overlay-outside {
            position: fixed;
            left: 0;
            top: 0;
            width: 100%;
            height: 100%;
            z-index: -1;
            background: $app-theme;
            opacity: 0;

            .hd-modal-open &{
                opacity: 0.8;
                transition: opacity 0.5s;
            }
        }

        .overlay-body {
            color: $info-color;
            font-size: $font-size-md;
            line-height: 27px;

            h5 {
                color: $mono-midnight;
            }
        }

        .overlay-header {
            margin-bottom: $lg-value;
            text-align: left;
        }

        .overlay-close {
            position: absolute;
            cursor: pointer;
            z-index: 1;
            top: 16px;
            right: 20px;
            @include media-breakpoint-up(md) {
                top: 35px;
                right: 32px;
            }
            > img {
                height: 28px;
                width: 28px;
                @include media-breakpoint-up(md) {
                    height: 35px;
                    width: 35px;
                }
            }
        }

        .overlay-close-multi {
            top: 56px;
            @include media-breakpoint-up(md) {
                top: 75px;
            }
        }
        .hd-modal-header {
            > span {
                position: absolute;
                z-index: 1;
                top: 16px;
                right: 20px;
                @include media-breakpoint-up(md) {
                    top: 35px;
                    right: 32px;
                }
                > img {
                    height: 28px;
                    width: 28px;
                    @include media-breakpoint-up(md) {
                        height: 35px;
                        width: 35px;
                    }
                }
            }
        }

        .overlay-footer {
            margin-top: 20px;
            @include media-breakpoint-up(md) {
              margin-top: 40px;
            }
        }

        .overlay-footer__cancel-btn {
          padding-top: 13px;
          @include media-breakpoint-up(sm) {
            padding-top: 0;
          }
        }
    }

    .error {
        margin-top: $sm-value;
    }
}
