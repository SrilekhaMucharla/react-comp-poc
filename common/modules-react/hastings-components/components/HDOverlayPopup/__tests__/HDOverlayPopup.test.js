import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import HDOverlayPopup from '../HDOverlayPopup';

configure({ adapter: new Adapter() });

describe('<HDOverLayPopup />', () => {
    it('render component with text only and sample content tree', () => {
        const tree = renderer.create(
            <HDOverlayPopup
                id="sample-id"
            >
                <p>Sample text</p>
            </HDOverlayPopup>
        )
            .toJSON();

        expect(tree)
            .toMatchSnapshot();
    });
});
