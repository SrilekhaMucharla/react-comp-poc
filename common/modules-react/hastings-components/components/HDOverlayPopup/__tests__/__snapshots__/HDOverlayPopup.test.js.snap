// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDOverLayPopup /> render component with text only and sample content tree 1`] = `
<div
  className="hd-information-tooltip "
>
  <div
    className="hd-overlay-btn "
    hidden={false}
    id="sample-id"
    onClick={[Function]}
  />
</div>
`;
